### PR TITLE
Save progress of propose-downstream and make it available via API

### LINKED ISSUE #1292: Save the progress of propose-downstream in database and make it available via API
To be able to show the progress of the `propose-downstream` job, we need to save it to the database and provide it via service API to show it on our dashboard (packit/dashboard#155).

Tasks:
- [ ] Suggest a change of the database schema (with the info we want to save), discuss with the team, update models and prepare migration.
- [ ] Save the progress during the process of `propose-downstream` (there can be multiple states we want to track).
- [ ] Provide an API endpoint for the `propose-downstream` jobs.

### PR DESCRIPTION
**Pull request by nikromen**
TODO:

- [x] suggest a change of database and discuss within the team
- [x] update models
- [x] prepare migration
- [x] save the progress during `propose-downstream`
- [x] provide an API endpoint
- [x] test models
- [x] test api

Fixes #1292 

---

TODO

### CODE REVIEW

**Comment by nikromen** [comment]
recheck

**Comment by nikromen** [comment]
recheck

**Comment by nikromen** [comment]
recheck

**Comment by nikromen** [comment]
recheck

**Comment by nikromen** [comment]
recheck

File: packit_service/models.py [hunk 1]
```diff
@@ -1600,8 +1616,168 @@ def __repr__(self):
...
+    error = "error"
+    retry = "retry"
+    submitted = "submitted"
+
+
+class ProposeTargetModel(ProjectAndTriggersConnector, Base):
+    __tablename__ = "propose_run_targets"
+    id = Column(Integer, primary_key=True)
+    branch = Column(String, default="unknown")
+    url = Column(String)
```
Comment (TomasTomecek) [inline packit_service/models.py:1631]: please clarify what this URL is

**Review by TomasTomecek — APPROVED** [review]
I mostly checked the schema, LGTM

**Review by lbarcziova — APPROVED** [review]
really nice changes! :rocket:

just a note - I got confused in some parts of the code with the naming for propose downstream as a whole, propose downstream for a target and corresponding statuses enums, it would be great if we could be consistent, e.g. ProposeDownstreamModel (with ProposeDownstreamStatus), ProposeDownstreamTargetModel (with ProposeDownstreamTarget status), WDYT?

but I don't want to hold you back because of this if you find the names clear, just wanted to avoid renaming in the future (since we had many issues with this recently :D)

**Comment by nikromen** [comment]
> it would be great if we could be consistent, e.g. ProposeDownstreamModel (with ProposeDownstreamStatus), ProposeDownstreamTargetModel (with ProposeDownstreamTarget status)

Yes, nice! Thank you, I'll apply your suggestions. I was struggling with the naming scheme and how I should name it.

**Pull request merged**